fondant/data_io.py

```diff
@@ -209,7 +207,11 @@ def create_subset_dataframe(
             subset_df = create_subset_dataframe(subset_name, subset_columns, df)
 
             # set index
-            subset_df.set_index("id")
+            if subset_df.index.name != "uid":
+                subset_df["uid"] = (
+                    subset_df["id"].astype("str") + "_" + subset_df["source"]
```

Comment:
I think concatenating `{source}_{id}` instead of `{id}_{source}` might be less costly, since chances are that the current order of the records resembles this more already.